packages/angular/cli/src/commands/mcp/tools/build.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import { z } from 'zod';
+import { execSync } from 'child_process';
+import { declareTool } from './tool-registry';
+
+const CONFIGURATIONS = {
+  development: {
+    args: '-c development',
+  },
+  production: {
+    args: '',
+  },
+};
+
+const BUILD_STATUSES = ['success', 'failure'] as const;
+type BuildStatus = (typeof BUILD_STATUSES)[number];
+
+const buildToolInputSchema = z.object({
+  project: z
+    .string()
+    .optional()
+    .describe(
+      'Which project to build in a monorepo context. If not provided, builds the top-level project.',
+    ),
+  configuration: z
+    .enum(Object.keys(CONFIGURATIONS) as [string, ...string[]])
+    .optional()
+    .describe('Which build configuration to use. Defaults to "development".'),
+});
+
+export type BuildToolInput = z.infer<typeof buildToolInputSchema>;
+
+const buildToolOutputSchema = z.object({
+  status: z.enum(BUILD_STATUSES).describe('Build status.'),
+  logs: z.string().describe('Output logs from `ng build`.'),
+  path: z.string().optional().describe('The output path the build project was written into.'),
+});
+
+export type BuildToolOutput = z.infer<typeof buildToolOutputSchema>;
+
+export function runBuild(input: BuildToolInput) {
+  const configurationName = input.configuration ?? 'development';
+  const configuration = CONFIGURATIONS[configurationName as keyof typeof CONFIGURATIONS];
+  const command = ['ng', 'build'];
+  if (input.project) {
+    command.push(input.project);
+  }
+  if (configuration.args) {
+    command.push(configuration.args);
+  }
+
+  let status: BuildStatus = 'success';
+  let logs = '';
+  let outputPath: string | undefined;
+
+  try {
+    logs = execSync(command.join(' ')).toString();
```

Comment:
Consider: I typically recommend `execFile` over `exec` (whether the sync or async variant) because it removes shell escaping. Using `exec` you need to properly quote arguments and run the risk of a malicious actor injecting some unexpected environment variables or something like that. `execFile` is much safer in that regard.